.github/workflows/ci.yml

```diff
@@ -21,8 +21,9 @@ jobs:
       - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
-          python -m pip install pre-commit
-          python -m pip freeze --local
+          python -m venv venv
+          source venv/bin/activate
+          pip install pre-commit
```

Comment:
why not using python -m?
why did you remove `python -m pip freeze --local`

Reply:
1. Does it really matter if I use `python -m` or `pip install` directly ?
2. Because it's irrelevant -- it freezes the current state of the virtual env such that it can be recreated somewhere else. We dont't need that here